fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -605,22 +608,12 @@ private <M extends Message> void updateSecondaryIndexes(@Nullable final FDBIndex
         for (Index index : indexes) {
             final IndexMaintainer maintainer = getIndexMaintainer(index);
             final CompletableFuture<Void> future;
-            if (!maintainer.isIdempotent() && isIndexWriteOnly(index)) {
-                // In this case, the index is still being built, so we are not
-                // going to update the record unless the rebuild job has already
-                // gotten to this range.
-                final Tuple primaryKey = newRecord == null ? oldRecord.getPrimaryKey() : newRecord.getPrimaryKey();
-                future = maintainer.addedRangeWithKey(primaryKey)
-                        .thenCompose(present -> {
-                            if (present) {
-                                return maintainer.update(oldRecord, newRecord);
-                            } else {
-                                return AsyncUtil.DONE;
-                            }
-                        });
-                if (!MoreAsyncUtil.isCompletedNormally(future)) {
-                    futures.add(future);
-                }
+            if (isIndexWriteOnly(index)) {
```

Comment:
There are other calls to `update`, do any of those need to be updated to do this check? They weren't doing the previous check, so they may have already been broken for non-idempotent use cases.

Reply:
Yeah, that seems to be the case. The other calls are:

1. Within the `TextIndexMaintainer`, which will only get called on (idempotent) text indexes, so we're fine
2. Within the indexer, which wants to update the index maintainer unconditionally
3. Within `FDBRecordStore::runSyntheticMaintainers`, which I'm less certain about, but that already was in a bit of a suspect state

So nothing seems to be updated in response to this, but maybe we should revisit this after this is in